module/spring-boot-neo4j/src/main/java/org/springframework/boot/neo4j/autoconfigure/Neo4jAutoConfiguration.java

```diff
@@ -73,10 +91,11 @@ PropertiesNeo4jConnectionDetails neo4jConnectionDetails(Neo4jProperties properti
 	@Bean
 	@ConditionalOnMissingBean
 	Driver neo4jDriver(Neo4jProperties properties, Environment environment, Neo4jConnectionDetails connectionDetails,
-			ObjectProvider<ConfigBuilderCustomizer> configBuilderCustomizers) {
+			ObjectProvider<ConfigBuilderCustomizer> configBuilderCustomizers,
+			ObjectProvider<ObservationRegistry> observationRegistryProvider) {
```

Comment:
We can't do this here. If we did then we could not configure a Neo4j driver without the observation registry on the classpath.

Reply:
Michael already answered the most bits. I think the approach with your `ConfigBuilderCustomiser` makes most sense. Thanks for putting the effort in to take it over the line ;)